pkg/lib/feature/forgotpassword/provider.go

```diff
@@ -284,17 +277,49 @@ func (p *Provider) ResetPasswordByCode(codeStr string, newPassword string) (err
 		return
 	}
 
+	return
+}
+
+func (p *Provider) CheckResetPasswordCode(codeStr string) error {
```

Comment:
I can now refresh as many times as I want to guess the reset password code.

Instead of rate limiting the attempt, we want to limit failed attempt. Rename `AntiBruteForceVerifyBucket` to `TrackFailedAttemptBucket`. Take the token when the code is invalid. No need to call RequireToken or CheckToken before verifying the code.

Reply:
If we don't check/require token before verifying the code, the result as seen by attacker would be either rate limit error or valid code? i.e. attacker could just ignore rate limit error and continue, since we don't check rate limit for correct code.

Reply:
Updated.